metadata-ingestion/src/datahub/ingestion/source/fs/s3_fs.py

```diff
@@ -0,0 +1,43 @@
+import boto3
+from datahub.ingestion.source.fs.fs_base import FileSystem, FileStatus
+from datahub.ingestion.source.fs.s3_list_iterator import S3ListIterator
+from urllib.parse import urlparse
+from typing import Iterable
+
+
+class S3FileSystem(FileSystem):
+    S3_FILE_SYSTEM_SCHEMA = "s3"
+    _s3 = boto3.client('s3')
+
+    def create(self, path: str):
+        raise NotImplementedError()
+
+    def open(self, path: str):
```

Comment:
i think this was is more readable and less error prune: 

    def open(self, path: str):
        bucket, key = self._parse_path(path)
        response = self._get_s3_object(bucket, key)
        self._assert_ok_status(response)
        return response['Body']

    def _parse_path(self, path: str):
        parsed = urlparse(path)
        return parsed.netloc, parsed.path.lstrip('/')